jerry-core/parser/regexp/re-compiler.c

```diff
@@ -472,7 +472,7 @@ re_find_bytecode_in_cache (ecma_string_t *pattern_str_p, /**< pattern string */
           && ecma_compare_ecma_strings (cached_pattern_str_p, pattern_str_p))
       {
         JERRY_DDLOG ("RegExp is found in cache\n");
-        return re_cache[*idx];
+        return (re_compiled_code_t *) re_cache[*idx];
```

Comment:
I would change re_cache to store re_compiled_code_t pointers. This is what is there anyway.